test/src/main/java/org/apache/accumulo/test/functional/TabletMergeabilityIT.java

```diff
@@ -0,0 +1,406 @@
+
+      // Wait for merge back to default tablet
+      Wait.waitFor(() -> hasExactTablets(getCluster().getServerContext(), tableId,
+          Set.of(new KeyExtent(tableId, null, null))), 10000, 200);
+    }
+  }
+
+  @Test
+  public void testMergeAfter() throws Exception {
+    String tableName = getUniqueNames(1)[0];
+    try (AccumuloClient c = Accumulo.newClient().from(getClientProps()).build()) {
+      c.tableOperations().create(tableName);
+      var tableId = TableId.of(c.tableOperations().tableIdMap().get(tableName));
+
+      TreeSet<Text> splits = new TreeSet<>();
+      splits.add(new Text(String.format("%09d", 333)));
+      splits.add(new Text(String.format("%09d", 666)));
+      splits.add(new Text(String.format("%09d", 999)));
+
+      var delay = Duration.ofSeconds(5);
+      var startTime = c.instanceOperations().getManagerTime();
+      c.tableOperations().putSplits(tableName, TabletMergeabilityUtil.splitsWithDefault(splits,
+          TabletMergeability.after(Duration.ofSeconds(5))));
+
+      Wait.waitFor(() -> countTablets(getCluster().getServerContext(), tableName, tm -> true) == 4,
+          5000, 200);
+
+      // Wait for merge back to default tablet
+      Wait.waitFor(() -> hasExactTablets(getCluster().getServerContext(), tableId,
+          Set.of(new KeyExtent(tableId, null, null))), 10000, 200);
+
+      var elapsed = c.instanceOperations().getManagerTime().minus(startTime);
+      assertTrue(elapsed.compareTo(delay) > 0);
+    }
+  }
+
+  @Test
+  public void testMergeabilityMaxFiles() throws Exception {
+    String tableName = getUniqueNames(1)[0];
+    try (AccumuloClient c = Accumulo.newClient().from(getClientProps()).build()) {
+      Map<String,String> props = new HashMap<>();
+      // disable compactions and set a low merge file max
+      props.put(Property.TABLE_MAJC_RATIO.getKey(), "9999");
+      props.put(Property.TABLE_MERGE_FILE_MAX.getKey(), "3");
+      c.tableOperations().create(tableName, new NewTableConfiguration().setProperties(props)
+          .withInitialTabletAvailability(TabletAvailability.HOSTED));
+      var tableId = TableId.of(c.tableOperations().tableIdMap().get(tableName));
+
+      // Create new tablets that won't merge automatically
+      SortedMap<Text,TabletMergeability> splits = new TreeMap<>();
+      for (int i = 500; i < 5000; i += 500) {
+        splits.put(row(i), TabletMergeability.never());
+      }
+      c.tableOperations().putSplits(tableName, splits);
+
+      // Verify we now have 10 tablets
+      Wait.waitFor(() -> countTablets(getCluster().getServerContext(), tableName, tm -> true) == 10,
+          5000, 500);
+
+      // Ingest data so tablet will split, each tablet will have several files because
+      // of the flush setting
+      VerifyParams params = new VerifyParams(getClientProps(), tableName, 5_000);
+      params.startRow = 0;
+      params.flushAfterRows = 100;
+      TestIngest.ingest(c, params);
+      VerifyIngest.verifyIngest(c, params);
+
```

Comment:
Could add a check of the number of files here.  Maybe it should do a waitFor instead of assertTrue though.

```
assertTrue(FileMetadataUtil.countFiles(getCluster().getServerContext(), tableName) > 3);
```